content/blog/2022/2022-01-25-annotation-processing.md

```diff
@@ -0,0 +1,951 @@
+date: 2022-01-27 00:00:00 +1100 
+modified: 2022-01-27 00:00:00 +1100
+authors: [syedaf]
+description: "An Introduction to Annotations and Annotation Processing"
+image: 
+url: 
+---
+
+## An Introduction to Annotations and Annotation Processing 
+
+An Annotation is a construct associated with Java source code elements such as classes, methods, and variables that provide information to the program at compile-time or run-time based on which the program can take further action. The Annotation Processor processes these Annotations at compile time to provide functionality such as code generation, error checking, etc.  
+
+The java.lang.annotation package provides the Core annotation functionality and also gives us the capability to create our Custom annotations that can be processed with Annotation Processors.
+
+In this article, we will discuss the topic of Annotations and demonstrate the power of Annotation Processing with a real-world example.
+
+## Annotation Basics
+
+An annotation is preceded by the @ sign. Some common examples of annotations are @Override and @Entity. These are Standard or built-in annotations provided by Java through the java.lang.annotations package. We can further extend the Core functionality to provide our custom annotations.
+
+An Annotation by itself does not perform any action on the program. It simply provides information about the program that can be used at compile-time or run-time to perform further processing. For eg., if we had a Parent and Child class as below:
+
+```java
+public class ParentClass {
+
+    public String getName() {..}
+}
+
+public class ChildClass extends ParentClass {
+
+    @Override
+    public String getname() {..}
+}
+```
+
+If we were to run this program without the @Override annotation we would not get any error since  'getname' would just be an additional method to 'getName' in ParentClass. By adding the @Override annotation in ChildClass the Compiler is able to enforce the rule that the overriding method in the child class should have the same case-sensitive name as that in the parent class, and so the program would throw an error at compile-time, thereby trapping an error which could have gone undetected even at run-time.
+
+## Code for this Article
+
+The code for the examples below can be downloaded at:
+
+
+
+## Standard Annotations
+
+Below are some of the most common Annotations in use. These are Standard annotations that Java provides as part of the java.lang.annotations package. To see their full effect it would be best to run the code snippets from the command line since most IDEs provide their custom options that alter warning levels.
+
+### @SuppressWarnings 
+
+**Use Case for @SuppressWarnings** - It is used to indicate that warnings on code compilation should be ignored. We may want to suppress warnings that clutter up the build output. @SuppressWarnings("unchecked") for example, suppresses warnings associated with raw types. For eg., when we run the following code:
```

Comment:
Please also go through the article and mark code fragments as `code`, like `@SuppressWarnings("unchecked")` in this case.